app/uk/gov/hmrc/fhregistrationfrontend/controllers/EnterOtherStoragePremisesController.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.fhregistrationfrontend.controllers
+
+import play.api.data.Form
+import play.api.data.Forms.tuple
+import play.api.mvc._
+import uk.gov.hmrc.fhregistrationfrontend.actions.Actions
+import uk.gov.hmrc.fhregistrationfrontend.config.FrontendAppConfig
+import uk.gov.hmrc.fhregistrationfrontend.forms.definitions.StoragePremisesForm
+import uk.gov.hmrc.fhregistrationfrontend.forms.mappings.Mappings.{skippingOnePrefix, yesOrNo}
+import uk.gov.hmrc.fhregistrationfrontend.forms.models.StoragePremise
+import uk.gov.hmrc.fhregistrationfrontend.views.Views
+import uk.gov.hmrc.fhregistrationfrontend.forms.navigation.Navigation.noNavigation
+import uk.gov.hmrc.fhregistrationfrontend.views.helpers.RepeatingPageParams
+
+import javax.inject.Inject
+
+class EnterOtherStoragePremisesController @Inject()(
+  ds: CommonPlayDependencies,
+  view: Views,
+  actions: Actions,
+  config: FrontendAppConfig)(
+  cc: MessagesControllerComponents
+) extends AppController(ds, cc) {
+
+  import actions._
+
+  val AddMoreKey = "addMore"
+  val ElementKey = "element"
+
+  val form: Form[(StoragePremise, Boolean)] = Form(
+    tuple(
+      ElementKey -> skippingOnePrefix(StoragePremisesForm.storagePremiseMapping),
+      AddMoreKey -> yesOrNo
+    )
+  )
+
+  val postAction = uk.gov.hmrc.fhregistrationfrontend.controllers.routes.EnterOtherStoragePremisesController.next()
+
+  def load(): Action[AnyContent] = userAction { implicit request =>
+    if (config.newBusinessPartnerPagesEnabled) {
+      Ok(view.storage_premise(form, noNavigation, "1", RepeatingPageParams(false, None), postAction))
```

Comment:
The user can select to repeat this page with the "Are there any more premises" question.
The current flow in live updates the url value based on this so we may need to edit this function and the routes to take in a query param.
This function can be modified to take in a query string param and passed in as the sectionId.
For the time being it can have a default value of "1" set  like below:
`def load(storagePremisesNum: String = "1")... = {
... 
Ok(view.storage_premise(form, noNavigation, storagePremisesNum, RepeatingPageParams(false, None), postAction))
...
}`